go/libkb/kbsig.go

```diff
@@ -32,14 +32,14 @@ func merkleRootInfo(g *GlobalContext) (ret *jsonw.Wrapper) {
 }
 
 type KeySection struct {
-	Key                GenericKey
-	EldestKID          keybase1.KID
-	ParentKID          keybase1.KID
-	HasRevSig          bool
-	RevSig             string
-	SigningUser        UserBasic
-	IncludePGPHash     bool
-	SharedDHGeneration *SharedDHGeneration
+	Key                   GenericKey
+	EldestKID             keybase1.KID
+	ParentKID             keybase1.KID
+	HasRevSig             bool
+	RevSig                string
+	SigningUser           UserBasic
+	IncludePGPHash        bool
+	SharedDHKeyGeneration *SharedDHKeyGeneration
```

Comment:
Do you feel strongly about `nil` versus `0` meaning unset here?  It might make code simpler to have non-pointers. You can't have a `0` generation for shared DH keys.